### ISSUE TITLE
DatePicker not highlighting today's date on initial click

### ISSUE THREAD

**Issue by MitchLindsay** [mgcrea/angular-strap#2185]
Hello!

When using the DatePicker, if I click today's date initially, it doesn't highlight. If I click another date first, **then** click today's date, it will highlight. I can reproduce this behavior in the DatePicker on the [demo page](http://mgcrea.github.io/angular-strap/#/datepickers).

I narrowed it down to this [else-if statement](https://github.com/mgcrea/angular-strap/blob/master/src/datepicker/datepicker.js#L532). If I remove the else-if conditional and change it _just_ an else statement, it works as I would expect it to.

```
update: function(date, force) {
  if (!this.built || force || date.getFullYear() !== viewDate.year || date.getMonth() !== viewDate.month) {
    angular.extend(viewDate, {year: picker.$date.getFullYear(), month: picker.$date.getMonth(), date: picker.$date.getDate()});
    picker.$build();
  else {
    viewDate.date = picker.$date.getDate();
    picker.$updateSelected();
  }
},
```

I assume the else-if is in place to prevent the same date from being selected twice, but update function is only called once if I click on a date that is already selected.

**Comment by ArslanWaqar** [comment]
So, when the date picker opens, today's date should be highlighted (which I think it still is) and it should be selected as first input, right?

**Comment by MitchLindsay** [comment]
Right, it's more of a UI bug. On the [demo page](http://mgcrea.github.io/angular-strap/#/datepickers), when you select today's date, the highlight color does not change from light blue to dark blue. But when you select another date, then select today's date again, _then_ it will correctly change the highlight color.
